Source/Core/Core/PowerPC/Jit64/Jit_Branch.cpp

```diff
@@ -237,16 +236,24 @@ void Jit64::bclrx(UGeckoInstruction inst)
 	if (inst.LK)
 		MOV(32, PPCSTATE_LR, Imm32(js.compilerPC + 4));
 
-	gpr.Flush(FLUSH_MAINTAIN_STATE);
-	fpr.Flush(FLUSH_MAINTAIN_STATE);
-	WriteBLRExit();
+	bool skip_exit = analyzer.HasOption(PPCAnalyst::PPCAnalyzer::OPTION_LEAF_INLINE) && js.op->isInlinedBLR;
+	if (!skip_exit)
+	{
+		gpr.Flush(FLUSH_MAINTAIN_STATE);
+		fpr.Flush(FLUSH_MAINTAIN_STATE);
+		WriteBLRExit();
+	}
+	else
+	{
+		MOV(32, PPCSTATE(pc), R(RSCRATCH));
+	}
 
 	if ((inst.BO & BO_DONT_CHECK_CONDITION) == 0)
 		SetJumpTarget(pConditionDontBranch);
 	if ((inst.BO & BO_DONT_DECREMENT_FLAG) == 0)
 		SetJumpTarget(pCTRDontBranch);
 
-	if (!analyzer.HasOption(PPCAnalyst::PPCAnalyzer::OPTION_CONDITIONAL_CONTINUE))
+	if (!analyzer.HasOption(PPCAnalyst::PPCAnalyzer::OPTION_CONDITIONAL_CONTINUE) && !skip_exit)
 	{
 		gpr.Flush();
```

Comment:
This touches the order of SetJumpTarget & gpr.Flush, so conditional continue will likely be broken, or slowed down.